### PR TITLE
Added a new abstraction for RendererList that will help transition to Render Graph and future RendererList API

### PR DESCRIPTION
**Pull request by JulienIgnace-Unity**
### Purpose of this PR
Added a new abstraction for RendererList that will help transition to Render Graph and future RendererList API

---
### Testing status
**Katana Tests**: [Running](https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=HDRP%2Fpre-rendergraph-refacto&automation-tools_branch=master&unity_branch=trunk)

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo
- [ ] C# and shader warnings (supress shader cache to see them)

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 1]
```diff
@@ -2341,143 +2341,109 @@ void RenderGizmos(CommandBuffer cmd, Camera camera, ScriptableRenderContext rend
...
+            Camera camera,
             ShaderTagId[] passNames,
             PerObjectData rendererConfiguration = 0,
-            RenderQueueRange? inRenderQueueRange = null,
+            RenderQueueRange? renderQueueRange = null,
             RenderStateBlock? stateBlock = null,
             Material overrideMaterial = null,
-            bool excludeMotionVector = false
-            )
+            bool excludeMotionVectorObjects = false
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: excludeObjectMotionVector

Comment (JulienIgnace-Unity) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: Fixed in latest commit

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 2]
```diff
@@ -2341,143 +2341,109 @@ void RenderGizmos(CommandBuffer cmd, Camera camera, ScriptableRenderContext rend
...
+            RenderQueueRange? renderQueueRange = null,
             RenderStateBlock? stateBlock = null,
-            Material overrideMaterial = null)
+            Material overrideMaterial = null,
+            bool excludeObjectMotionVectors = false
+        )
         {
-            m_SinglePassName[0] = passName;
-            RenderOpaqueRenderList(cull, hdCamera, renderContext, cmd, m_SinglePassName, rendererConfiguration, inRenderQueueRange, stateBlock, overrideMaterial);
+            return RendererList.Create(new RendererListDesc(passName, cull, camera)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: for a later PR, are we sure we want to do a new here? shouldn't this be pooled ?

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 3]
```diff
@@ -2560,49 +2529,35 @@ bool RenderDepthPrepass(CullingResults cull, HDCamera hdCamera, ScriptableRender
...
+                    // First deferred alpha tested materials. Alpha tested object have always a prepass even if enableDepthPrepassWithDeferredRendering is disabled
+                    var partialPrepassRenderQueueRange = new RenderQueueRange { lowerBound = (int)RenderQueue.AlphaTest, upperBound = (int)RenderQueue.GeometryLast - 1 };
 
-                            // First deferred material
-                            RenderOpaqueRenderList(cull, hdCamera, renderContext, cmd, m_DepthOnlyPassNames, 0, HDRenderQueue.k_RenderQueue_AllOpaque, excludeMotionVector: excludeMotion);
+                    // First deferred material
+                    var rendererList1 = CreateOpaqueRendererList(
+                        cull, hdCamera.camera, m_DepthOnlyPassNames,
+                        renderQueueRange: fullDeferredPrepass ? HDRenderQueue.k_RenderQueue_AllOpaque : partialPrepassRenderQueueRange,
+                        excludeObjectMotionVectors: objectMotionEnabled);
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: this is warong, it should used excludeMotion

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 4]
```diff
@@ -2560,49 +2529,35 @@ bool RenderDepthPrepass(CullingResults cull, HDCamera hdCamera, ScriptableRender
...
 
-                            HDUtils.SetRenderTarget(cmd, m_SharedRTManager.GetPrepassBuffersRTI(hdCamera.frameSettings), m_SharedRTManager.GetDepthStencilBuffer());
+                    // Then forward only material that output normal buffer
+                    var rendererList2 = CreateOpaqueRendererList(cull, hdCamera.camera, m_DepthForwardOnlyPassNames, excludeObjectMotionVectors: objectMotionEnabled);
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: same here, use excludeMotion

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs (hunk 2 above)
Comment (JulienIgnace-Unity) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: It's a struct, no garbage here.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs (hunk 3 above)
Comment (JulienIgnace-Unity) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: Fixed in latest commit

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs (hunk 4 above)
Comment (JulienIgnace-Unity) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: Fixed in latest commit

**Pull request merged**